### PR TITLE
Apply strictNullChecks to src/components/views/location/* (#10249

### PR DESCRIPTION
**Pull request by kerryarchibald**
Fixes https://github.com/vector-im/element-web/issues/24655

## Checklist

-   [ ] Tests written for new code (and old code if feasible)
-   [ ] Linter and other CI checks pass
-   [ ] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: src/components/views/elements/AccessibleButton.tsx [hunk 1]
```diff
@@ -75,7 +75,7 @@ type IProps<T extends keyof JSX.IntrinsicElements> = DynamicHtmlElementProps<T>
     disabled?: boolean;
     className?: string;
     triggerOnMouseDown?: boolean;
-    onClick: ((e: ButtonEvent) => void | Promise<void>) | null;
+    onClick?: ((e: ButtonEvent) => void | Promise<void>) | null;
```
Comment (t3chguy) [inline src/components/views/elements/AccessibleButton.tsx:78]: I think null here might be intentional for the caller to demonstrate intent

**Review by t3chguy — APPROVED** [review]
Looks sane otherwise, tyvm!!

File: src/components/views/elements/AccessibleButton.tsx (hunk 1 above)
Comment (kerryarchibald) [reply src/components/views/elements/AccessibleButton.tsx]: Makes sense, removed this change.

**Comment by t3chguy** [comment]
Bypassing sonar due to this being a tsc strict PR

**Pull request merged**